src/base/vesting/Index.svelte

```diff
@@ -1,52 +1,83 @@
 <script lang="ts">
-  import { State, state } from "./state";
-  import { getInfo, withdrawVested } from "./vesting";
-  import type { VestingInfo } from "./vesting";
   import type { Session } from "@app/session";
   import type { Config } from "@app/config";
-  import Modal from "@app/Modal.svelte";
+  import type { VestingInfo } from "./vesting";
+
+  import * as utils from "@app/utils";
   import Address from "@app/Address.svelte";
-  import { formatAddress, isAddressEqual } from "@app/utils";
   import Button from "@app/Button.svelte";
+  import Modal from "@app/Modal.svelte";
   import TextInput from "@app/TextInput.svelte";
+  import { state, getInfo, withdrawVested } from "./vesting";
 
   export let config: Config;
   export let session: Session | null;
 
   let contractAddress = "";
   let info: VestingInfo | null = null;
+  let validationMessage: string | undefined = undefined;
+  let valid: boolean = false;
 
   async function loadContract(config: Config) {
-    state.set(State.Loading);
-    info = await getInfo(contractAddress, config);
-    state.set(State.Idle);
+    if (!valid) {
+      return;
+    }
+
+    $state = "loading";
+    try {
+      info = await getInfo(contractAddress, config);
+    } catch (error) {
+      validationMessage =
+        "Couldn't load contract, check dev console for details.";
+      console.error(error);
+    }
+    $state = "idle";
   }
 
   function reset() {
     info = null;
-    state.set(State.Idle);
+    $state = "idle";
```

Comment:
Can we use the `state.set("idle")` syntax? I think it reads more clearly that something is happening, than using the magic syntax.

Reply:
Hmm. I'll change it to `set` here, so we can merge it. But I wanna think a bit more about it and revisit this topic at some later point.

2f8d0332bd89e63c28544e024006693b74848025